### PR TITLE
Updated example to use frontend/backend structure and renamed gateway app to frontend

### PR DESCRIPTION
**Pull request by sshver**
*Issue #, if available:*

*Description of changes:*

1. `gateway` in the walkthrough is an application that can’t be replicated with a Virtual Gateway. Renamed `gateway` app to `frontend` as used in other examples of the walkthrough.
  
Walkthrough and examples updated:
 - ColorApp
 - fargate
 - howto-cross-account
 - howto-servicediscovery-cloudmap
 - howto-timeout-policy
 - howto-tls-file-provided

2. AWS ECR CLI command `get-login` is deprecated in AWS CLI version 2. Replaced `get-login` to use `get-login-password` for CLI version 2.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by rajal-amzn — CHANGES_REQUESTED** [review]
Instead of renaming the gateway to frontend, we would want to change the gateway task from an application with gateway image to do the forward to a Virtual Gateway.

File: walkthroughs/howto-cross-account/primary-account/deploy.sh [hunk 1]
```diff
@@ -12,13 +12,20 @@ fi
...
 deploy_image() {
-    echo "Deploying Gateway image to ECR..."
-    aws ecr describe-repositories --repository-name ${PROJECT_NAME}/gateway >/dev/null 2>&1 || aws ecr create-repository --repository-name ${PROJECT_NAME}/gateway
-    docker build -t ${ECR_IMAGE_PREFIX}/gateway ${DIR}/gateway --build-arg BACKEND_SERVICE=backend.${PROJECT_NAME}.local
-    $(aws --profile ${AWS_PROFILE} ecr get-login --no-include-email)
-    docker push ${ECR_IMAGE_PREFIX}/gateway
+    echo "Deploying Frontend image to ECR..."
+    aws ecr describe-repositories --repository-name ${PROJECT_NAME}/frontend >/dev/null 2>&1 || aws ecr create-repository --repository-name ${PROJECT_NAME}/frontend
+    docker build -t ${ECR_IMAGE_PREFIX}/frontend ${DIR}/frontend --build-arg BACKEND_SERVICE=backend.${PROJECT_NAME}.local
+    if [ $AWS_CLI_VERSION -gt 1 ]; then
```
Comment (dfezzie) [inline walkthroughs/howto-cross-account/primary-account/deploy.sh:22]: Appreciate this change

**Comment by dfezzie** [comment]
I agree with changing the names from `gateway` to `frontend` as it helps avoid confusion with the `gateway` feature. I agree with @rajal-amzn though and we should actually utilize the App Mesh `gateway` feature in these examples. I would start with one to get the approach down and then open separate PRs for the different walk throughs.

**Comment by sshver** [comment]
Closing this pull request. Created separate PRs for each walkthrough.

**Pull request closed without merging**